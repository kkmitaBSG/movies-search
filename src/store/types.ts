export const api_key = "92b418e837b833be308bbfb1fb2aca1e";

export enum MoviesActionType {
  getMovies = "GET_MOVIES",
  getMoviesFail = "GET_MOVIES_FAIL",
  getSearchMovies = "GET_SEARCH_MOVIES",
  getMoviesNextPage = "GET_MOVIES_NEXT_PAGE",
  getMovieDetails = "GET_MOVIE_DETAILS",
}

export interface Movie {
  id: string;
  title: string;
  overview: string;
  poster_path?: string;
  poster?: string;
}

export interface Movies {
  results?: Movie[];
  page: number;
}

export interface MovieDetails {
  id?: string;
  backdrop_path?: string;
  title?: string;
  backdrop?: string;
  overview?: string;
  production_countries?: Array<{ name: string }>;
  release_date?: string;
  genres?: Array<{ name: string }>;
}

export interface MovieDetailsState {
  [id: string]: MovieDetails;
}

export interface State {
  movies: Movies;
  movieDetails: MovieDetails;
}

export const moviePosterOriginalPath = "https://image.tmdb.org/t/p/w500";
