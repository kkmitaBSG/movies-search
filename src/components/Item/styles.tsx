import styled from "styled-components";
import { NavLink } from "react-router-dom";

export const Item = styled(NavLink)<{ poster?: string }>`
  flex: 1 1 auto;
  margin: 10px;
  box-shadow: 0 0 10px #737373;
  border-radius: 10px;
  width: 250px;
  height: 450px;
  cursor: pointer;
  overflow: hidden;
  text-decoration: none;
  color: #FFF;

  ${({ poster }) => poster && `background-image: url(${poster});`}
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center center;
`;

export const AdditionalInfo = styled.div`
  opacity: 0;
  background-color: #00000090;
  height: 100%;
  padding: 50px 20px 0;
  border-radius: 10px;
  transition: opacity 0.3s ease-out;

  ${Item}:hover & {
    opacity: 1;
  }
`;

export const SubElem = styled.div`
  display: -webkit-box;
  -webkit-line-clamp: 13;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const Name = styled.div`
  color: #ffbc9e;
  margin-bottom: 10px;
  font-size: 25px;
  font-weight: 500;
  text-align: center;
`;
