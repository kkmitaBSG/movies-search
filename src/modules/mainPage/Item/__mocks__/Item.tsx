import React, { ReactElement } from "react";
import { Items } from "../../../../api/useFetch";

const MockedComponent = (item: Items): ReactElement => (
  <React.Fragment key={item.id}>ItemMockedComponent</React.Fragment>
);

export default MockedComponent;
