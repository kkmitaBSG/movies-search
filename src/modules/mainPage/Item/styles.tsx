import styled from "styled-components";

export const Item = styled.div`
  flex: 1 1 auto;
  margin: 10px;
  padding: 10px;
  box-shadow: 0 0 10px #aaa;
  border-radius: 10px;
`;

export const SubElem = styled.div``;

export const Name = styled(SubElem)`
  color: #0025aa;
  font-weight: 500;
`;

export const Desc = styled.span`
  font-size: 0.9em;
  color: #00ac84;
`;
