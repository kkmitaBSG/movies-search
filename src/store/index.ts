import { configureStore } from "@reduxjs/toolkit";
import moviesReducer from "./reducers/moviesReducer";
import movieDetailsReducer from "./reducers/movieDetailsReducer";
import createSagaMiddleware from "redux-saga";
import rootSaga from "./sagas";

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: {
    movies: moviesReducer,
    movieDetails: movieDetailsReducer,
  },
  middleware: [sagaMiddleware],
});

sagaMiddleware.run(rootSaga);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
