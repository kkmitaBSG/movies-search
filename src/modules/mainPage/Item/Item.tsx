import React from "react";
import { Items } from "../../../api/useFetch";
import messages from "../../../constants/messages";
import * as S from "./styles";

const Item = React.memo(({ title, overview, original_language }: Items) => (
    <S.Item>
      <S.Name>{title}</S.Name>
      {overview && (
        <S.SubElem>
          <S.Desc>{messages.overviewDesc}</S.Desc> {overview}
        </S.SubElem>
      )}
      {original_language && (
        <S.SubElem>
          <S.Desc>{messages.originalLanguageDesc}</S.Desc> {original_language}
        </S.SubElem>
      )}
    </S.Item>
  )
);

export default Item;
