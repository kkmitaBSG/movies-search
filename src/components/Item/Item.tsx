import React from "react";
import { Movie } from "../../store/types";
import * as S from "./styles";

const Item = React.memo(({ id, title, overview, poster }: Movie) => (
  <S.Item to={`details/${id}`} poster={poster}>
    <S.AdditionalInfo>
      <S.Name>{title}</S.Name>
      <S.SubElem>{overview}</S.SubElem>
    </S.AdditionalInfo>
  </S.Item>
));

export default Item;
