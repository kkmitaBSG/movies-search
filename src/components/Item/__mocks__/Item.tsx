import React, { ReactElement } from "react";
import { Movie } from "../../../store/types";

const MockedComponent = (movie: Movie): ReactElement => (
  <React.Fragment key={movie.id}>ItemMockedComponent</React.Fragment>
);

export default MockedComponent;
