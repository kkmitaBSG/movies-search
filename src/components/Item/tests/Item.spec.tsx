import Item from "../Item";
import renderer from "react-test-renderer";
import { Movie } from "../../../store/types";
import { MemoryRouter } from "react-router-dom";

const defaultProps: Movie = {
  id: "1",
  title: "item1Name",
  overview: "overview1",
  poster: "xyz.eu",
};

describe("Item", () => {
  test("with full filled props", () => {
    const component = renderer.create(
      <MemoryRouter>
        <Item {...defaultProps} />
      </MemoryRouter>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
