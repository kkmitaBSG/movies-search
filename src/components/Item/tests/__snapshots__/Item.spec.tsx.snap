// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item with full filled props 1`] = `
<a
  className="sc-bcXHqe hlXchW"
  href="/details/1"
  onClick={[Function]}
  poster="xyz.eu"
>
  <div
    className="sc-gswNZR gmVPuo"
  >
    <div
      className="sc-hLBbgP lFNHr"
    >
      item1Name
    </div>
    <div
      className="sc-dkrFOg emALgx"
    >
      overview1
    </div>
  </div>
</a>
`;
